import_service/lambda_func/import_file_parser.ts

```diff
@@ -45,65 +56,78 @@ export const importFileParserHandler = async (event: any) => {
 }
 
 export const importFileParserService = async (key: string) => {
-	try {
-		console.log(key)
-		const client = new S3Client({})
+	const client = new S3Client({})
+	const sqsClient = new SQSClient()
 
-		const fileName = key.split("/").at(-1)
+	const fileName = key.split("/").at(-1)
 
-		if (!fileName) {
-			throw new Error("Missing file name")
-		}
+	if (!fileName) {
+		throw new Error("Missing file name")
+	}
 
-		const srcParams = {
-			Bucket: BUCKET_NAME,
-			Key: key,
-			ContentType: "text/csv",
-		}
+	const srcParams = {
+		Bucket: BUCKET_NAME,
+		Key: key,
+		ContentType: "text/csv",
+	}
 
-		const destParams = {
-			...srcParams,
-			Key: `parsed/${fileName}`,
-			CopySource: `${BUCKET_NAME}/${key}`,
-		}
+	console.log(`S3 source parameters: ${JSON.stringify(srcParams)}`)
 
-		const getCommand = new GetObjectCommand(srcParams)
-		const copyCommand = new CopyObjectCommand(destParams)
-		const deleteCommand = new DeleteObjectCommand(srcParams)
+	const destParams = {
+		...srcParams,
+		Key: `parsed/${fileName}`,
+		CopySource: `${BUCKET_NAME}/${key}`,
+	}
 
-		const { Body: stream } = await client.send(getCommand)
+	console.log(`S3 destination parameters: ${JSON.stringify(destParams)}`)
 
-		if (!stream) {
-			throw new Error("Stream error")
-		}
+	const getCommand = new GetObjectCommand(srcParams)
+
+	const { Body: stream } = await client.send(getCommand)
 
-		return new Promise<string>((resolve, reject) => {
-			;(stream as Readable)
-				.pipe(csvParser())
-				.on("data", (chunk) => {
-					console.log("CSV chunk:", chunk)
-				})
-				.on("end", async () => {
-					const message = "CSV file was parsed successfully!"
-					console.log(message)
-
-					try {
-						await client.send(copyCommand)
-						console.log("The file was copied to parsed folder")
-						await client.send(deleteCommand)
-						console.log("The file was deleted from uploaded folder")
-
-						return resolve(message)
-					} catch (e) {
-						reject(e)
-					}
-				})
-				.on("error", (err: any) => {
-					console.error(err)
-					reject(err)
-				})
-		}).then((msg) => msg)
-	} catch (err: any) {
-		throw new Error(err)
+	console.log(`Stream retrieved from S3: ${stream}`)
+
+	if (!stream) {
+		throw new Error("Stream error")
 	}
+
+	return new Promise<string>((resolve, reject) => {
+		;(stream as Readable)
+			.pipe(csvParser())
+			.on("data", async (chunk: any) => {
+				console.log("CSV chunk:", chunk)
+				const messageParams = {
+					QueueUrl: process.env.SQS_URL,
+					MessageBody: JSON.stringify(chunk),
+				}
+				try {
+					console.log("Message params", messageParams)
+					await sqsClient.send(new SendMessageCommand(messageParams))
```

Comment:
Parsing the next chunk of products data is blocked by sending a message to SQS, it would be better to separate products parsing and sending messages to SQS.